### PR TITLE
[Kotlin Migration] CollectionTest

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 1]
```diff
@@ -13,71 +13,51 @@
...
-    public void beforeUploadClosesCollection() {
-        Collection col = getCol();
-        assertThat("db should be open", CollectionHelper.getInstance().colIsOpen(), is(true));
-        col.beforeUpload();
-        assertThat("db should be closed", CollectionHelper.getInstance().colIsOpen(), is(false));
+    fun beforeUploadClosesCollection() {
+        val col = col
+        MatcherAssert.assertThat("db should be open", CollectionHelper.getInstance().colIsOpen(), Matchers.`is`(true))
+        col.beforeUpload()
+        MatcherAssert.assertThat("db should be closed", CollectionHelper.getInstance().colIsOpen(), Matchers.`is`(false))
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:55]: For discussion: Given that tests are much less likely to be broken, I feel it's worth fixing `MatcherAssert/Assert etc...` having imports removed, and changing `is` to something valid for Kotlin (`equalTo`, maybe `eq` is defined)

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/kotlinMigration.gradle [hunk 2]
```diff
@@ -42,8 +42,8 @@ permission notice:
 
 // Example of class name: "/com/ichi2/anki/UIUtils.kt"
 // Ensure that it starts with '/' (slash)
-def source = Source.MAIN
-def className = ""
+def source = Source.TEST
+def className = "com/ichi2/libanki/CollectionTest.kt"
```
Comment (david-allison-1) [inline AnkiDroid/kotlinMigration.gradle]: `/`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt (hunk 1 above)
Comment (Akshay0701) [reply AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt]: I agree, did it

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 3]
```diff
@@ -13,71 +13,52 @@
...
-    public void beforeUploadClosesCollection() {
-        Collection col = getCol();
-        assertThat("db should be open", CollectionHelper.getInstance().colIsOpen(), is(true));
-        col.beforeUpload();
-        assertThat("db should be closed", CollectionHelper.getInstance().colIsOpen(), is(false));
+    fun beforeUploadClosesCollection() {
+        val col = col
+        MatcherAssert.assertThat("db should be open", CollectionHelper.getInstance().colIsOpen(), equalTo(true))
+        col.beforeUpload()
+        MatcherAssert.assertThat("db should be closed", CollectionHelper.getInstance().colIsOpen(), equalTo(false))
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:56]: `MatcherAssert` import

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 4]
```diff
@@ -112,126 +93,122 @@ public class CollectionTest extends RobolectricTest {
...
-        mm.save(m, true); // todo: remove true which is not upstream
-        assertEquals(2, col.cardCount());
+        val m = col.models.current()
+        val mm = col.models
+        val t = Models.newTemplate("Reverse")
+        t.put("qfmt", "{{Back}}")
+        t.put("afmt", "{{Front}}")
+        mm.addTemplateModChanged(m!!, t)
+        mm.save(m, true) // todo: remove true which is not upstream
+        Assert.assertEquals(2, col.cardCount().toLong())
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:112]: `Assert` import

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 5]
```diff
@@ -112,126 +93,122 @@ public class CollectionTest extends RobolectricTest {
...
-        note.setItem("Front", "three");
-        note.setItem("Back", "four");
-        n = col.addNote(note);
-        assertEquals(2, n);
-        assertEquals(4, col.cardCount());
+        note = col.newNote()
+        note.setItem("Front", "three")
+        note.setItem("Back", "four")
+        n = col.addNote(note)
+        Assert.assertEquals(2, n.toLong())
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:118]: `toLong()` shouldn't be needed

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 6]
```diff
@@ -112,126 +93,122 @@ public class CollectionTest extends RobolectricTest {
...
+        Assert.assertEquals(4, col.cardCount().toLong())
         // check q/a generation
-        Card c0 = note.cards().get(0);
-        assertThat(c0.q(), containsString("three"));
+        val c0 = note.cards()[0]
+        MatcherAssert.assertThat(c0.q(), Matchers.containsString("three"))
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:122]: `Matchers` import

File: AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt [hunk 7]
```diff
@@ -112,126 +93,122 @@ public class CollectionTest extends RobolectricTest {
...
-        note.setItem("Back", "new2");
-        col.addNote(note);
-        assertEquals(0xc2a6b03f, col.getDb().queryLongScalar("select csum from notes"));
+    fun test_fieldChecksum() {
+        val col = col
+        val note = col.newNote()
+        note.setItem("Front", "new")
+        note.setItem("Back", "new2")
+        col.addNote(note)
+        Assert.assertEquals(-0x3d594fc1, col.db.queryLongScalar("select csum from notes"))
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:143]: ouch! Can we revert this to `0xc2a6b03f`?

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

**Comment by david-allison-1** [comment]
````
com.ichi2.libanki.CollectionTest > test_timestamps FAILED
    java.lang.AssertionError: expected: java.lang.Long<5> but was: java.lang.Integer<5>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at com.ichi2.libanki.CollectionTest.test_timestamps(CollectionTest.kt:176)
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
This looks beautiful, consider this an approve once the test passes

**Comment by Akshay0701** [comment]
removed some more toLong, hope this time it works,

I can't able to check this test locally, because i always get javaheap error, i tried every other solution to solve this but same java heap error occurring while running the tests, i assume its my ram issue...

**Comment by david-allison-1** [comment]
> removed some more toLong, hope this time it works,
> 
> I can't able to check this test locally, because i always get javaheap error, i tried every other solution to solve this but same java heap error occurring while running the tests, i assume its my ram issue...

Could you mention this in Discord. We've seen this problem before and it should be investigated.

**Comment by mikehardy** [comment]
```

com.ichi2.libanki.CollectionTest > test_timestamps FAILED
    java.lang.AssertionError: expected: java.lang.Long<105> but was: java.lang.Integer<105>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at com.ichi2.libanki.CollectionTest.test_timestamps(CollectionTest.kt:180)
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Didn't submit this one, should be easy to fix

**Review by mikehardy — APPROVED** [review]
I can't see any comments here that are not addressed
running CI tasks locally after rebase to make sure

**Comment by mikehardy** [comment]
Wow, this is from *January* :sweat_smile: :hourglass_flowing_sand:

**Pull request merged**